gradflow.html

```diff
@@ -66,37 +66,35 @@ <h1 class="form-title">CREATE COURSE</h1>
                   <ol class="form-input-wrapper">
                      <ol class="form-input-item">
                         <label for="c-prefixnum"><b>Prefix Number</b></label>
-                        <input type="text" placeholder="Enter Prefix Number" name="c-prefixnum" id="c-prefixnum-create"
-                           required>
+                        <input type="text" placeholder="Enter Prefix Number" name="c-prefixnum" id="c-prefixnum-create" pattern="^[A-Za-z]{3,4}\s?[0-9]{3}[A-Za-z]?$" title="3-4 letters, optional space, 3 digits, optional letter. ex: ABCD 123A" required>
```

Comment:
This regex does not accept 3 digit numbers as input. This is the format of the vast majority of prefix numbers. Seems to be an issue with the grouping of on the optional components. Try something more like this:
```suggestion
                        <input type="text" placeholder="Enter Prefix Number" name="c-prefixnum" id="c-prefixnum-create" pattern="^([A-Za-z]{3,4}\s)?[0-9]{3}([A-Za-z])?$" title="optional 3-4 letters + space, 3 digits, optional letter. ex: ABCD 123A" required>
```

Reply:
do you mean entering just 123 instead of CPSC 123? I was only going to accept things of the form 
**ABC 123** or    **ABC123** or    **ABC 123E** or   **ABC123E** or
**ABCD 123** or **ABCD123** or **ABCD 123E** or **ABCD123E**

_(also all the variations of capitilization such as **abc 123**, etc. )_

but  I am open to just accepting 3 digits as well no problem
**123**

Reply:
I decided against validating prereqs because the user can enter other prereqs besides classes. 
for example cpsc 362's prereqs could be: 

_CPCS 131, fulfillment of upper-division writing requirement, cs major_

but i think i can validate prereqs! it will just make our prereq inputs more strict and we assume the user will only enter class prerequisites.

Reply:
Oh I see what you mean on prereqs. Hmm, not sure if that would be better put in description. The script will waste time looking for a node call "fulfillment of upper-division writing requirement," which obliviously will not exist since that is an invalid prefixNumber. 

In regards to the prefix number, yes, I would like it to accept plain three-digit numbers alongside numbers with a major code prefix.